### ISSUE TITLE
[bug] Channel Regex for Namespace cuased Null Pointer Exception

### ISSUE THREAD

**Issue by deefdragon** [centrifugal/centrifugo#759]
**Describe the bug**

Setting channel_regex on a namespace caused the following NPE

<details>
<summary>Click to expand stacktrace</summary>

```
goroutine 121 [running]:
regexp.(*Regexp).doExecute(0x0?, {0x0?, 0x0?}, {0x0?, 0x0?, 0x0?}, {0xc000024131?, 0x0?}, 0x0?, 0x0, ...)
	/opt/hostedtoolcache/go/1.21.5/x64/src/regexp/exec.go:527 +0x89
regexp.(*Regexp).doMatch(...)
	/opt/hostedtoolcache/go/1.21.5/x64/src/regexp/exec.go:514
regexp.(*Regexp).MatchString(...)
	/opt/hostedtoolcache/go/1.21.5/x64/src/regexp/regexp.go:533
github.com/centrifugal/centrifugo/v5/internal/client.(*Handler).validChannelName(0x1?, {0xc000024131?, 0x1?}, {0x0, 0x0, 0x0, 0x1, 0x6fc23ac00, 0x0, 0x0, ...}, ...)
	/home/runner/work/centrifugo/centrifugo/internal/client/handler.go:601 +0x70
github.com/centrifugal/centrifugo/v5/internal/client.(*Handler).validateChannelName(0xc000bd6dc0, {0x1fe7c38, 0xc0000d43c0}, {0xc000024131?, 0xc0001e9230?}, {0x0, 0x0, 0x0, 0x1, 0x6fc23ac00, ...}, ...)
	/home/runner/work/centrifugo/centrifugo/internal/client/handler.go:611 +0x8d
github.com/centrifugal/centrifugo/v5/internal/client.(*Handler).OnSubscribe(0xc000bd6dc0, {0x1fe7c38, 0xc0000d43c0}, {{0xc000024120, 0x15}, {0x0, 0x0}, {0x0, 0x0, 0x0}, ...}, ...)
	/home/runner/work/centrifugo/centrifugo/internal/client/handler.go:644 +0x436
github.com/centrifugal/centrifugo/v5/internal/client.(*Handler).Setup.func2.3.1()
	/home/runner/work/centrifugo/centrifugo/internal/client/handler.go:199 +0x8f
github.com/centrifugal/centrifugo/v5/internal/client.(*Handler).runConcurrentlyIfNeeded(0x40?, {0x1fe12f0?, 0xc000020000?}, 0x100000?, 0x7fc4741264c0?, 0x7fc4bacee108?)
	/home/runner/work/centrifugo/centrifugo/internal/client/handler.go:286 +0xfb
github.com/centrifugal/centrifugo/v5/internal/client.(*Handler).Setup.func2.3({{0xc000024120, 0x15}, {0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, 0x0, 0x0}, ...)
	/home/runner/work/centrifugo/centrifugo/internal/client/handler.go:198 +0x16f
github.com/centrifugal/centrifuge.(*Client).handleSubscribe(0xc0000d43c0, 0xc00011e870, 0xc00011e7e0, {0x1623020?, 0xc000bdc040?, 0x2cac300?}, 0x0)
	/home/runner/go/pkg/mod/github.com/centrifugal/centrifuge@v0.30.7-0.20231225122811-902c7cf8701f/client.go:1645 +0x23e
github.com/centrifugal/centrifuge.(*Client).dispatchCommand(0xc0000d43c0, 0xc00011e7e0, 0x38)
	/home/runner/go/pkg/mod/github.com/centrifugal/centrifuge@v0.30.7-0.20231225122811-902c7cf8701f/client.go:1234 +0x6c5
github.com/centrifugal/centrifuge.(*Client).HandleCommand(0xc0000d43c0, 0xc00011e7e0, 0x7fc473df9b40?)
	/home/runner/go/pkg/mod/github.com/centrifugal/centrifuge@v0.30.7-0.20231225122811-902c7cf8701f/client.go:1074 +0xc5
github.com/centrifugal/centrifuge.HandleReadFrame(0xc0000d43c0, {0x7fc473df9b40, 0xc0000f0038})
	/home/runner/go/pkg/mod/github.com/centrifugal/centrifuge@v0.30.7-0.20231225122811-902c7cf8701f/handler_websocket.go:240 +0x148
github.com/centrifugal/centrifuge.(*WebsocketHandler).ServeHTTP.func2()
	/home/runner/go/pkg/mod/github.com/centrifugal/centrifuge@v0.30.7-0.20231225122811-902c7cf8701f/handler_websocket.go:205 +0x5cf
created by github.com/centrifugal/centrifuge.(*WebsocketHandler).ServeHTTP in goroutine 119
	/home/runner/go/pkg/mod/github.com/centrifugal/centrifuge@v0.30.7-0.20231225122811-902c7cf8701f/handler_websocket.go:165 +0x5db
```
</details>

**Versions**

Centrifugo version is 5.2.0
Client library used is centrifuge-js of version 5.0.1
Operating system is kubernetes

**Steps to Reproduce**

using the following config.json

```json
{
  "admin": true,
  "allow_anonymous_connect_without_token": true,
  "allowed_origins": [...],
  "engine": "redis",
  "namespaces": [
    {
      "allow_history_for_anonymous": true,
      "allow_history_for_client": true,
      "allow_history_for_subscriber": true,
      "allow_subscribe_for_anonymous": true,
      "allow_subscribe_for_client": true,
      "channel_regex": "^[0-9]+$",
      "force_recovery": true,
      "history_size": 1,
      "history_ttl": "30s",
      "name": "name",
      "presence": false
    }
  ],
  "redis_prefix": "",
  "token_ecdsa_public_key": "FOO"
}
```

and subscribing to `name:1234` on the client results in the server NPE. This kicks the client from the server.

removing the regex option causes the connection to work fine.

**Expected behavior**

The subscription is allowed to proceed, and invalid subscriptions are ignored as expected.

**Comment by FZambia** [comment]
@deefdragon hello, thanks a lot for the detailed description, it will be fixed by #760 in the next release

**Comment by deefdragon** [comment]
Given you merged in the fix, I think it is ok to close.
